components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessor.java

```diff
@@ -0,0 +1,635 @@
+     * @return boolean if validation is successful.
+     */
+    private boolean isSidClaimExists(JWTClaimsSet claimsSet) {
+
+        return StringUtils.isNotBlank((String) claimsSet.getClaim(OIDCAuthenticatorConstants.Claim.SID));
+    }
+
+    /**
+     * Do the sub claim validation.
+     *
+     * @param claimsSet
+     * @return boolean if validation is successful.
+     */
+    private boolean isSubClaimExists(JWTClaimsSet claimsSet) {
+
+        return StringUtils.isNotBlank(claimsSet.getSubject());
+    }
+
+    /**
+     * Do the event claim validation according to OIDC back-channel logout specification.
+     *
+     * @param event
+     * @return boolean if validation is successful.
+     * @throws LogoutClientException
+     */
+    private void validateEventClaim(JSONObject event) throws LogoutClientException {
+
+        String eventClaimValue = event.getAsString(OIDCAuthenticatorConstants.Claim.BACKCHANNEL_LOGOUT_EVENT);
+        if (event == null ||
+                !StringUtils.equals(eventClaimValue, OIDCAuthenticatorConstants.Claim.BACKCHANNEL_LOGOUT_EVENT_CLAIM)) {
+            throw new LogoutClientException(ErrorMessages.LOGOUT_TOKEN_EVENT_CLAIM_VALIDATION_FAILED.getCode(),
+                    ErrorMessages.LOGOUT_TOKEN_EVENT_CLAIM_VALIDATION_FAILED.getMessage());
+        }
+    }
+
+    /**
+     * Do the nonce claim validation according to OIDC back-channel logout specification.
+     *
+     * @param claimsSet
+     * @return boolean if validation is successful.
+     * @throws LogoutClientException
+     */
+    private void validateNonce(JWTClaimsSet claimsSet) throws LogoutClientException {
+
+        if (StringUtils.isNotBlank((String) claimsSet.getClaim(OIDCAuthenticatorConstants.Claim.NONCE))) {
+            throw new LogoutClientException(ErrorMessages.LOGOUT_TOKEN_NONCE_CLAIM_VALIDATION_FAILED.getCode(),
+                    ErrorMessages.LOGOUT_TOKEN_NONCE_CLAIM_VALIDATION_FAILED.getMessage());
+        }
+    }
+
+    /**
+     * Get the OIDC authenticator configs from the xml file.
+     *
+     * @return AuthenticatorConfig.
+     */
+    protected AuthenticatorConfig getAuthenticatorConfig() {
+
+        AuthenticatorConfig authConfig = FileBasedConfigurationBuilder.getInstance()
+                .getAuthenticatorBean(AUTHENTICATOR_NAME);
+        if (authConfig == null) {
+            authConfig = new AuthenticatorConfig();
+            authConfig.setParameterMap(new HashMap<>());
+        }
+        return authConfig;
+    }
+
+    @Override
+    public String getCallbackPath(IdentityMessageContext context) {
+
+        return null;
+    }
+
+    @Override
+    public String getRelyingPartyId() {
+
+        return null;
+    }
+
+    @Override
+    public String getRelyingPartyId(IdentityMessageContext context) {
+
+        return null;
+    }
+
+    @Override
+    public boolean canHandle(IdentityRequest identityRequest) {
+
+        boolean canHandle = false;
+        if (identityRequest != null) {
```

Comment:
Shall we revert this condition? That way code will be more readable and we won't have to indent the code unnecessarily.